src/SprykerEco/Zed/Payone/Business/Payment/Hook/PostSaveHook.php

```diff
@@ -124,7 +124,7 @@ public function executeCheckoutPostSaveHook(
      */
     protected function checkApiLogRedirectAndError(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse): CheckoutResponseTransfer
     {
-        $apiLogTransfer = $this->payoneRepository->findLastApiLogByOrderId($quoteTransfer->getPayment()->getPayone()->getFkSalesOrder());
+        $apiLogTransfer = $this->payoneRepository->findLastApiLogByOrderId((int)$quoteTransfer->getPayment()->getPayone()->getFkSalesOrder());
```

Comment:
If you don't cast them null is not a valid type for this method I think. And getorFail is not an option, because in line 142 there is a proper error code returned. If you use getOrFail here, instead of checking if the result is present, you get a hard error instead of the proper error code before it even reaches that point.

Reply:
Currently, this line throws an exception for `null`, since `findLastApiLogByOrderId` expects int. Code execution just won’t make it to line 142 anyway. So by using `OrFail` here we won’t introduce a new exception.